File: Programmers/구명 보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    que = deque(sorted(people))
  
    while que:
        if(len(que) == 1):
            answer += 1
            break
          
        if(que[0]+que[-1]) <= limit:
            que.pop()
            que.popleft()
            answer += 1
        else:
            que.pop()
            answer += 1
          
    return answer

File: Programmers/구명 보트.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int solution(int people[], int people_count, int limit) {
    int answer = 0;
  
    qsort(people, people_count, sizeof(int), compare);

    int left = 0;              
    int right = people_count - 1;

    while (left <= right) {
        if (left == right) {
            answer++;
            break;
        }
        if (people[left] + people[right] <= limit) {
            left++;
            right--;
        } else {
            right--;
        }
        answer++;
    }

    return answer;
}
